File: front-end/src/app/service-models/app/v1/client/payroll-transaction-s-m.ts
import { CoinManagementServiceModelBase } from '../../base/coin-management-service-model-base';
import { PaymentModeSM } from '../../enums/payment-mode-s-m.enum';
import { PaymentTypeSM } from '../../enums/payment-type-s-m.enum';

export class PayrollTransactionSM extends CoinManagementServiceModelBase<number> {
    paymentAmount!: number;
    paymentFor!: string;
    paymentMode!: PaymentModeSM;
    paymentType!: PaymentTypeSM;
    toPay!: number;
    toPaid!: number;
    paymentPaid!: boolean;
    clientUserId!: number;
    clientEmployeeCTCDetailId!: number;
    clientCompanyDetailId!: number;
    errorInGeneration!: boolean;
    errorMessage!: string;
}

File: backend/payroll-backend/Components/Siffrum.Web.Payroll.ServiceModels/v1/Client/PayrollTransactionSM.cs
using Siffrum.Web.Payroll.ServiceModels.Base;
using Siffrum.Web.Payroll.ServiceModels.Enums;

namespace Siffrum.Web.Payroll.ServiceModels.v1.Client
{
    public class PayrollTransactionSM : SiffrumPayrollServiceModelBase<int>
    {
        public float PaymentAmount { get; set; }
        public string PaymentFor { get; set; }
        public PaymentModeSM PaymentMode { get; set; }
        public PaymentTypeSM PaymentType { get; set; }
        public float ToPay { get; set; }
        public float ToPaid { get; set; }
        public bool PaymentPaid { get; set; }
        public int ClientUserId { get; set; }
        public int ClientEmployeeCTCDetailId { get; set; }
        public int ClientCompanyDetailId { get; set; }
        public bool ErrorInGeneration { get; set; }
        public string ErrorMessage { get; set; }

    }
}
